src/linker/Linker/UnconditionalSuppressMessageAttributeState.cs

```diff
@@ -99,10 +106,127 @@ private static bool TryDecodeSuppressMessageAttributeData (CustomAttribute attri
 
 		private bool IsLocallySuppressed (int id, ICustomAttributeProvider provider, out SuppressMessageInfo info)
 		{
-			Dictionary<int, SuppressMessageInfo> suppressions;
 			info = default;
-			return _localSuppressions.TryGetValue (provider, out suppressions) &&
+			if (provider == null)
+				return false;
+
+			return _localSuppressions.TryGetValue (provider, out var suppressions) &&
 				suppressions.TryGetValue (id, out info);
 		}
+
+		private bool IsGloballySuppressed (int id, ICustomAttributeProvider provider, out SuppressMessageInfo info)
+		{
+			DecodeGlobalSuppressMessageAttributes (provider);
+			return _globalSuppressions.HasGlobalSuppressions (id, provider, out info);
+		}
+
+		private void DecodeGlobalSuppressMessageAttributes (ICustomAttributeProvider provider)
+		{
+			ModuleDefinition module;
+			switch (provider.MetadataToken.TokenType) {
+			case TokenType.Module:
```

Comment:
I would extract this to a static helper - GetModuleForProvider - or something like that.
Then use it here and also in the parent class when we check for local suppressions on module/assembly. It would make the code more readable.